packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -1777,10 +1802,16 @@ function resetChildLanes(completedWork: Fiber) {
 
 function commitRoot(root) {
   const renderPriorityLevel = getCurrentPriorityLevel();
-  runWithPriority(
-    ImmediateSchedulerPriority,
-    commitRootImpl.bind(null, root, renderPriorityLevel),
-  );
+  const previousLanePriority = getCurrentLanePriority();
+  try {
+    setCurrentLanePriority(SyncLanePriority);
```

Comment:
Move this into `commitRoot`. There's only one place that throws, at the very end, so the try/finally isn't needed.